enum.c

```diff
@@ -871,49 +875,141 @@ ary_inject_op(VALUE ary, VALUE init, VALUE op)
- *     enum.reduce(initial, sym) -> obj
- *     enum.reduce(sym)          -> obj
- *     enum.reduce(initial) { |memo, obj| block }  -> obj
- *     enum.reduce          { |memo, obj| block }  -> obj
- *
- *  Combines all elements of <i>enum</i> by applying a binary
- *  operation, specified by a block or a symbol that names a
- *  method or operator.
- *
- *  The <i>inject</i> and <i>reduce</i> methods are aliases. There
- *  is no performance benefit to either.
- *
- *  If you specify a block, then for each element in <i>enum</i>
- *  the block is passed an accumulator value (<i>memo</i>) and the element.
- *  If you specify a symbol instead, then each element in the collection
- *  will be passed to the named method of <i>memo</i>.
- *  In either case, the result becomes the new value for <i>memo</i>.
- *  At the end of the iteration, the final value of <i>memo</i> is the
- *  return value for the method.
- *
- *  If you do not explicitly specify an <i>initial</i> value for <i>memo</i>,
- *  then the first element of collection is used as the initial value
- *  of <i>memo</i>.
- *
- *
- *     # Sum some numbers
- *     (5..10).reduce(:+)                             #=> 45
- *     # Same using a block and inject
- *     (5..10).inject { |sum, n| sum + n }            #=> 45
- *     # Multiply some numbers
- *     (5..10).reduce(1, :*)                          #=> 151200
- *     # Same using a block
- *     (5..10).inject(1) { |product, n| product * n } #=> 151200
- *     # find the longest word
- *     longest = %w{ cat sheep bear }.inject do |memo, word|
- *        memo.length > word.length ? memo : word
- *     end
- *     longest                                        #=> "sheep"
+ *     inject(symbol) -> obj
+ *     inject(initial_operand, symbol) -> obj
+ *     inject {|memo, element| ... } -> obj
+ *     inject(initial_operand) {|memo, element| ... } -> obj
+ *
+ *  Combines each successive element of +self+ with an accumulator.
+ *
+ *  A simple example of each of the above calling sequences:
+ *
+ *    # Sum of elements.
+ *    (1..4).inject(:+)                      # => 10
+ *    # Sum elements added to 2.
+ *    (1..4).inject(2, :+)                   # => 12
+ *    # Sum of squares of elements.
+ *    (1..4).inject {|sum, n| sum + n*n }    # => 30
+ *    # Sum of squares of elements added to 2.
+ *    (1..4).inject(2) {|sum, n| sum + n*n } # => 32
+ *
+ *  <b>Case 1: <tt>inject(symbol) -> obj</tt></b>
+ *
+ *  Given the single argument +symbol+ (which names a binary method) and no block:
+ *
+ *  - Combines elements of +self+ using the binary method:
+ *
+ *    - The first two operands are <tt>self[0]</tt> and <tt>self[1]</tt>.
+ *    - The next two operands are that result and <tt>self[2]</tt>.
+ *    - The next two operands are that result and <tt>self[3]</tt>.
+ *    - And so on.
+ *
+ *  - Returns the last result.
+ *
+ *  The returned object may not be of the same class as the first element.
```

Comment:
While true, I don't think we need to mention the return value may not be of the same class as the first element (or any element).  Mentioning it for only some cases and not others is misleading, since this is true of all cases (both cases when a symbol is used and when a block is used).